src/argilla/server/search_engine/commons.py

```diff
@@ -270,42 +270,53 @@ async def similarity_search(
         vector_settings: VectorSettings,
         value: Optional[List[float]] = None,
         record: Optional[Record] = None,
+        query: Optional[Union[TextQuery, str]] = None,
         user_response_status_filter: Optional[UserResponseStatusFilter] = None,
-        max_results: conint(ge=2, le=500) = 100,
+        metadata_filters: Optional[List[MetadataFilter]] = None,
+        max_results: int = 100,
+        order: Union[Literal["most_similar"], Literal["least_similar"]] = "most_similar",
         threshold: Optional[float] = None,
     ) -> SearchResponses:
-        if not (value or record):
-            raise ValueError("Must provide vector value or record to compute the similarity search")
+        if bool(value) == bool(record):
+            raise ValueError("Must provide either vector value or record to compute the similarity search")
 
         vector_value = value
+        record_id = None
 
         if not vector_value:
             for vector in record.vectors:
                 if vector.vector_settings_id == vector_settings.id:
                     vector_value = vector.value
+                    record_id = record.id
+                    break
 
         if not vector_value:
             raise ValueError("Cannot find a vector value to apply with provided info")
 
+        if order == "least_similar":
+            for i in range(0, len(vector_value)):
+                vector_value[i] *= -1
```

Comment:
maybe we can put this in a function called `inverse_vector`

Reply:
Done